azurerm/internal/services/redis/tests/resource_arm_redis_linked_server_test.go

```diff
@@ -0,0 +1,177 @@
+		CheckDestroy: testCheckAzureRMRedisLinkedServerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMRedisLinkedServer_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMRedisLinkedServerExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMRedisLinkedServer_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_redis_linked_server", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMRedisLinkedServerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMRedisLinkedServer_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMRedisLinkedServerExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMRedisLinkedServer_requiresImport),
+		},
+	})
+}
+
+func testCheckAzureRMRedisLinkedServerExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Redis.LinkedServerClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %q", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		cacheName := rs.Primary.Attributes["redis_cache_name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := client.Get(ctx, resourceGroup, cacheName, name)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Linked Server %q (cache %q resource group: %q) does not exist", name, cacheName, resourceGroup)
+			}
+			return fmt.Errorf("Bad: Get on redis.LinkedServersClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMRedisLinkedServerDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Redis.LinkedServerClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_redis_linked_server" {
+			continue
+		}
+
+		redisCacheName := rs.Primary.Attributes["redis_cache_name"]
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := client.Get(ctx, resourceGroup, redisCacheName, name)
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Linked Server still exists:\n%#v", resp)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMRedisLinkedServer_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "pri" {
+  name     = "accpriRG-%d"
```

Comment:
we always use acctestRG
```suggestion
  name     = "acctestRG-redis-%d"
```